module-2/backend/week_one.md

```diff
@@ -5,30 +5,79 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+
 11. What's responsive design?
+Responsive design allows the web page to respond to screen size.
+
 12. What is CRUD and why is it important?
+Create, Read, Update, Destroy - these are the main functionality for all web applications. You can create, read, update, or destroy any given thing in an application. 
+
 13. What does HTTP stand for? 
+Hyper Text Transfer Protocol
+
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+<%= %> - this one will be viewed in the HTML displayed on the website
+<% %> - this one will not be viewed in the HTML
 15. What's an ORM?
+Object relational mapper - it is a wrapper written in an object oriented language like Ruby or Python that allows for interaction with a database (such as SQL). In this case, each row (attribute) in a database corresponds to a object. Those rows can be manipulated and used just like regular objects. From a design perspective, this is especially beneficial because it pushes all state management to the database. The ORM simply writes SQL in such a way that we can interact with the database using objects.  
 16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
+ActiveRecord
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+/restaurants - GET - shows all restaurants - READ
+/restaurants/:id - GET - views specific restaurant - READ
+/restaurants/new - GET - allows viewing of form to create new restaurant - CREATE
+/restaurants/new - POST - allows creation through form submission of new restaurant - CREATE
+/restaurants/edit - GET - allows viewing of existing restaurants, can click to edit - UPDATE
+/restaurants/edit/:id - POST - allows editing of a specific restaurant - UPDATE
+/restaurants/edit/:id - DELETE - allows deleting of specific object - DELETE
+
 18. What's a migration? 
+A migration is a way of changing and evolving the structure of the database. It is created by ActiveRecord in order to change, create, or update the database. 
 19. When you create a migration, does it automatically modify your database?
+No. It creates a migration method that can be customized to modify the database. You have to run the migration file to impact the schema of the database.
+
 20. How does a model relate to a database?
+This is confusing to me, because it seems that it serves two purposes. First, a model corresponds to a table and thus also defines relationships to other tables. However, in defining a table, the model also defines how that tables attributes (rows) will be structured, meaning how the instances of that object will created and managed. 

```

Comment:
I would say that the *migration* defines the table. The *model* makes use of the attributes of a table in the database to allow us to interact with it.